File: core/src/main/java/org/dockbox/darwin/core/events/AbstractCancellableEvent.java
package org.dockbox.darwin.core.events;

import org.dockbox.darwin.core.objects.events.Cancellable;

public abstract class AbstractCancellableEvent implements Cancellable {

    private boolean isCancelled;

    public AbstractCancellableEvent() {}

    @Override
    public boolean isCancelled() {
        return this.isCancelled;
    }

    @Override
    public void setCancelled(boolean cancelled) {
        this.isCancelled = cancelled;
    }
}

File: core/src/main/kotlin/org/dockbox/darwin/core/events/AbstractCancellableEvent.kt
package org.dockbox.darwin.core.events

import org.dockbox.darwin.core.objects.events.Cancellable

abstract class AbstractCancellableEvent : Cancellable {
    private var isCancelled = false
    override fun isCancelled(): Boolean {
        return isCancelled
    }

    override fun setCancelled(cancelled: Boolean) {
        isCancelled = cancelled
    }
}
